tests/resources/jormungandr/node.config

```diff
@@ -0,0 +1,13 @@
+storage: "/tmp/storage"
+logger:
+  verbosity: 1
+  format: json
+rest:
+  listen: "127.0.0.1:8443"
+  prefix: "api"
+peer_2_peer:
+  trusted_peers: []
+  public_access: "/ip4/127.0.0.1/tcp/8080"
+  topics_of_interests:
+    messages: low
+    blocks: normal
```

Comment:
We need to find a way to generate this document depending on the needs.

One easy way we can do that is by preparing the default configuration file in a _format ready way_.

NODE_CONFIG.yaml

```yaml
storage: {storage}
logger:
  verbosity: 1
  format: json
rest:
  listen: "{rest_listen}"
  prefix: "{rest_api}"
peer_2_peer:
  trusted_peers: {p2p_trusted_peers}
  public_access: "{p2p_public_access}"
  topics_of_interests:
    messages: {topic_message_interest_level}
    blocks: {topic_blocks_interest_level}
```

Then we can generate the configuration file for every nodes we need to setup:

```rust
    writeln!(output_file, include_str!("NODE_CONFIG.yaml"),
        storage = config.storage.unwrap_or("~"),
        rest_listen = config.rest.listen,
        rest_prefix = config.rest.prefix,
        p2p_trusted_peers = config.p2p.peers,
        p2p_public_access = config.p2p.public,
        topic_messages_interest = config.p2p.message_interest,
        topic_blocks_interest = config.p2p.blocks_interest,
   )
```

This will allow to call the function above with pre-defined parameters, to setup more easily the node(s) and how the topology will look like.

Ideally, we will need to do the same with the `genesis.yaml` file for the blockchain. and the `node_secret.yaml`  file too for the nodes we need to be able to write blocks in the blockchain.

Reply:
Absolutely agree. At this moment i hardcoded a few things (like node.config and gensis.yaml), but i will generate them on the fly in the nearest future as you proposed